frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx

```diff
@@ -0,0 +1,155 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { k8sCreate, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import {
+  ButtonBar,
+  history,
+  resourceObjPath,
+  RequestSizeInput,
+} from '@console/internal/components/utils';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { NooBaaObjectBucketClaimModel } from '@console/noobaa-storage-plugin/src/models';
+import { match } from 'react-router';
+import { commonReducer, defaultState } from '../object-bucket-page/state';
+
+export const CreateOBCPage: React.FC<CreateOBCPageProps> = (props) => {
+  const [state, dispatch] = React.useReducer(commonReducer, defaultState);
+
+  const namespace = _.get(props, 'match.params.ns');
+  const dropDownUnits = {
+    MiB: 'MiB',
+    GiB: 'GiB',
+    TiB: 'TiB',
+  };
+
+  React.useEffect(() => {
+    const obj: K8sResourceKind = {
+      apiVersion: `${NooBaaObjectBucketClaimModel.apiGroup}/${
+        NooBaaObjectBucketClaimModel.apiVersion
+      }`,
+      kind: NooBaaObjectBucketClaimModel.kind,
+      metadata: {
+        namespace,
+      },
+      spec: {
+        ssl: false,
+      },
+    };
+    if (state.scName) {
+      obj.spec.storageClassName = state.scName;
+    }
+    if (state.name) {
+      obj.metadata.name = state.name;
+      obj.spec.generateBucketName = state.name;
+    } else {
+      obj.metadata.generateName = 'true';
+      obj.metadata.generateBucketName = true;
+    }
+    dispatch({ type: 'setPayload', payload: obj });
+  }, [namespace, state.name, state.scName]);
+
+  const save = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    dispatch({ type: 'setProgress' });
+    k8sCreate(NooBaaObjectBucketClaimModel, state.payload)
+      .then((resource) => {
+        dispatch({ type: 'unsetProgress' });
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      })
+      .catch((err) => {
+        dispatch({ type: 'setError', message: err.message });
+        dispatch({ type: 'unsetProgress' });
+      });
+  };
+
+  return (
+    <div className="co-m-pane__body co-m-pane__form">
+      <Helmet>
+        <title>Create Object Bucket Claim</title>
+      </Helmet>
+      <h1 className="co-m-pane__heading co-m-pane__heading--baseline">
+        <div className="co-m-pane__name">Create Object Bucket Claim</div>
+        <div className="co-m-pane__heading-link">
+          <Link to={`/k8s/ns/${namespace}/objectbucketclaims/~new`} id="yaml-link" replace>
+            Edit YAML
```

Comment:
@bipuladh I don't think the `id` is being used. Please remove that.

Reply:
ack